src/zenml/materializers/base_materializer.py

```diff
@@ -326,3 +329,67 @@ def _extract_base_metadata(self, data: Any) -> Dict[str, "MetadataType"]:
         if isinstance(storage_size, int):
             return {"storage_size": StorageSize(storage_size)}
         return {}
+
+    @contextlib.contextmanager
+    def get_temporary_directory(
+        self,
+        delete_at_exit: bool,
+        delete_after_step_execution: bool = True,
+    ) -> Iterator[str]:
+        """Context manager to get a temporary directory.
+
+        Args:
+            delete_at_exit: If set to True, the temporary directory will be
+                deleted after the context manager exits.
+            delete_after_step_execution: If `delete_at_exit` is set to False and
+                this is set to True, the temporary directory will be deleted
+                after the step finished executing. If a materializer is being
+                used outside of the context of a step execution, the temporary
+                directory will not be deleted and the user is responsible for
+                deleting it themselves.
+
+        Yields:
+            Path to the temporary directory.
+        """
+        temp_dir = tempfile.mkdtemp(prefix="zenml-")
+
+        if delete_after_step_execution and not delete_at_exit:
```

Comment:
So, if I get it correctly, if you set `delete_at_exit` to `True`, setting `delete_after_step_execution` to `True` or `False` will not have any effect on how it is being cleaned up. I was initially worried that both being `True` would lead us to try to delete the same directory twice and eventually a `FileNotFoundError` but I guess this is not the case here.